src/views/Profile/PublicProfile.tsx

```diff
@@ -95,18 +104,25 @@ const PublicProfile = () => {
             <Flex alignItems={['start', null, 'center']} flexDirection={['column', null, 'row']}>
               <EditProfileAvatar profile={profile} />
               <Content>
-                <Username>{`@${profile.username}`}</Username>
+                <Flex alignItems="center">
+                  <Username>@{usernameVisibilityToggled ? username : username.replace(/./g, '*')}</Username>
+                  {usernameVisibilityToggled ? (
+                    <VisibilityOff ml="4px" onClick={toggleUsernameVisibility} cursor="pointer" />
```

Comment:
in order to avoid duplicated code you can do 
```
const Icon = usernameVisibilityToggled ? VisibilityOff : VisibilityOn
```

and then remove this ternary expressions

Reply:
Good idea, fixed with https://github.com/pancakeswap/pancake-frontend/pull/943/commits/f34743ff2ad4ff5dde61d846d474a30f4f41eca7